sourced/ml/cmd/args.py

```diff
@@ -17,6 +25,30 @@ def _get_help_string(self, action):
         return super()._get_help_string(action)
 
 
+def handle_input_arg(input_arg: str,
+                     filter_arg: Optional[str] = DEFAULT_FILTER_ARG,
+                     log: Optional[logging.Logger] = None):
+    """
+    Process input arguments and return an iterator over input files.
+
+    :param input_arg: Directory path to scan or `-` to get file paths from stdin
+    :param filter_arg: File name glob selector for input directory. Ignored if `input_arg` is \
+        equal to `-`.
+    :param log: Logger if you want to log handling process.
+    :return: an iterator over input files.
+    """
+    log = log.info if log else (lambda *x: None)
+    if input_arg == "-":
+        log("Reading file paths from stdin.")
+        for line in sys.stdin:
+            yield line.strip()
+    else:
+        log("Scanning %s", input_arg)
+        if filter_arg:
+            for path in Path(input_arg).glob(filter_arg):
```

Comment:
That is,

```python
for arg in input_arg:
    if path.isfile(arg):
        yield arg
    elif filter_arg:
        for path in Path(arg).glob(filter_arg):
            yield str(path)
```

Reply:
I want to process directories without filtering also, so I change it to:
```python
        for arg in input_arg:
            if os.path.isfile(arg):
                yield arg
            else:
                if filter_arg:
                    for path in Path(arg).glob(filter_arg):
                        yield str(path)
                elif os.path.isdir(arg):
                    for path in os.listdir(arg):
                        yield path
```

Reply:
See how complex it became?

This is why we need to nuke all this code and leave

```python
for arg in input_arg:
    yield arg
```

The rest is GNU find's job. I override you here, sorry, but it really must be done this way.